### DISCUSSION TITLE
Is mqtt_client work with python as well?

### DISCUSSION THREAD

**Discussion by RangeOfGlitching** [ika-rwth-aachen/mqtt_client#10] [Q&A] [answered]
Is mqtt_client also compatible with python and Cpp as mqtt_bridge?

**Comment by lreiher** [comment]
Please specify what exactly you mean by "compatible with Python and C++". The mqtt_bridge is a ROS Python Node, our mqtt_client is a ROS C++ Node(let).

Are you asking whether it is possible to exchange data between mqtt_bridge and mqtt_client?

**Reply by RangeOfGlitching** [reply]
HI @lreiher 
Please accept my apologies for my unclear question. I am a rookie at ROS and English
Using MQTT bridge, I can exchange ROS messages via MQTT broker. Can I do the same using MQTT_client?
for example
(python publish -> mqtt broker -> CPP subcribe  || CPP publish -> mqtt broker -> Python subcribe)

Thanks!

**Reply by lreiher** [reply, marked as answer]
When referring to MQTT bridge, are you specifically talking about the Python package [mqtt_bridge](https://github.com/groove-x/mqtt_bridge) or are you talking about a custom ROS Python node with the [paho-mqtt](https://pypi.org/project/paho-mqtt/) MQTT Python API?

I believe it is not possible to exchange messages between mqtt_bridge <> MQTT Broker <> mqtt_client. Both sides use different ways of serializing the ROS messages to MQTT messages. It could become possible if https://github.com/ika-rwth-aachen/mqtt_client/issues/9 is solved and JSON serialization is used on both sides.

Talking about any kind of MQTT client, e.g. Python MQTT API <> MQTT Broker <> mqtt_client, it is possible to exchange primitive messages, see [here](https://github.com/ika-rwth-aachen/mqtt_client#primitive-messages).
